typing/printtyp.ml

```diff
@@ -1486,6 +1506,11 @@ let unification_error unif tr txt1 ppf txt2 =
         txt2 (type_expansion t2) t2'
         (trace false "is not compatible with type") tr
         (explanation unif mis);
+      if not (is_Tvar t1 || is_Tvar t2)
```

Comment:
When explain the type error in the following, 
```
type 'a t
(fun (x : 'a t as 'a) -> x)
```
`t` is somehow not available in `env`.
I've added condition on `is_Tvar` to workaround this, BUT
 - I don't know if it is correct
 - if so, I don't know why it is correct